satpy/composites/__init__.py

```diff
@@ -1082,28 +1103,33 @@ def _get_and_sharpen_rgb_data_arrays_and_meta(self, datasets, optional_datasets)
             if 'rows_per_scan' in high_res.attrs:
                 new_attrs.setdefault('rows_per_scan', high_res.attrs['rows_per_scan'])
             new_attrs.setdefault('resolution', high_res.attrs['resolution'])
-            low_res_colors = ['red', 'green', 'blue']
-            low_resolution_index = low_res_colors.index(self.high_resolution_color)
+
         else:
             LOG.debug("No sharpening band specified for ratio sharpening")
             high_res = None
-            low_resolution_index = 0
 
         if high_res is not None:
```

Comment:
This condition and the `else` could be swapped to clean up the indentation. So `if high_res is None: return low_res_red, ...` and then no `else`, but instead dedent the version with the high resolution. Or you could extract the `bands = {...}` definition outside the if statement and do `return bands['red'], ...` outside of the `if` statement.

I'm just trying to think of ways to make CodeScene happier about the code quality.